src/main/java/org/gridsuite/network/map/dto/definition/extension/BusbarSectionFinderTraverser.java

```diff
@@ -6,42 +6,123 @@
  */
 package org.gridsuite.network.map.dto.definition.extension;
 
-import com.powsybl.iidm.network.IdentifiableType;
-import com.powsybl.iidm.network.Switch;
-import com.powsybl.iidm.network.Terminal;
-import com.powsybl.math.graph.TraverseResult;
+import com.powsybl.iidm.network.*;
+
+import java.util.*;
 
 /**
- * @author Slimane Amar <slimane.amar at rte-france.com>
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
  */
-public class BusbarSectionFinderTraverser implements Terminal.TopologyTraverser {
+// TODO : to remove when this class is available in network-store
+public final class BusbarSectionFinderTraverser {
+
+    private BusbarSectionFinderTraverser() {
+        throw new UnsupportedOperationException();
+    }
 
-    private final boolean onlyConnectedBbs;
+    private record NodePath(int node, List<SwitchInfo> pathSwitches, SwitchInfo lastSwitch) { }
```

Comment:
```suggestion
    private record Path(int startNode, List<SwitchInfo> traversedSwitches, SwitchInfo lastSwitch) { }
```
Something like this ?

Reply:
The term 'Path' is a generic name
it is ok for startNode and traversedSwitches